frontend/src/metabase/nav/containers/NewButton.jsx

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable react/prop-types */
+import React from "react";
+import { connect } from "react-redux";
+import { t } from "ttag";
+import EntityMenu from "metabase/components/EntityMenu";
+import Icon from "metabase/components/Icon";
+import Link from "metabase/core/components/Link";
+import { color, darken } from "metabase/lib/colors";
+import * as Urls from "metabase/lib/urls";
+
+const MODAL_NEW_DASHBOARD = "MODAL_NEW_DASHBOARD";
+const MODAL_NEW_COLLECTION = "MODAL_NEW_COLLECTION";
+
+import { getUser } from "../selectors";
+import {
+  getHasDataAccess,
+  getHasNativeWrite,
+  getPlainNativeQuery,
+  getHasDbWithJsonEngine,
+} from "metabase/new_query/selectors";
+
+function NewButton({
+  hasDataAccess,
+  hasNativeWrite,
+  hasDbWithJsonEngine,
+  setModal,
+}) {
+  return (
+    <EntityMenu
+      className="hide sm-show mr1 ml-auto"
+      trigger={
+        <Link
+          mr={1}
+          p={1}
+          hover={{
+            backgroundColor: darken(color("brand")),
+          }}
+          className="flex align-center rounded transition-background"
+          data-metabase-event={`NavBar;Create Menu Click`}
+        >
+          <Icon name="add" size={14} />
+          <h4 className="hide sm-show ml1 text-nowrap">{t`New`}</h4>
+        </Link>
```

Comment:
We're trying to use styled-components (`emotion`) for styling as much as possible. I can help to refactor that later, but please prefer it for new components 🙏

Reply:
I got tired of having to move a bunch of state selectors and imports to move one button around so I wrapped the existing "New" button entity menu in its own component. Is the rule of thumb that I should do a refactor of something like this when extracting it? If so happy to do it.

Reply:
Oh, gotcha, sorry for a false alarm. Let's skip it then!